src/libraries/System.Runtime.Numerics/src/System/Numerics/BigInteger.cs

```diff
@@ -1591,6 +1592,41 @@ public string ToString([StringSyntax(StringSyntaxAttribute.NumericFormat)] strin
             return BigNumber.FormatBigInteger(this, format, NumberFormatInfo.GetInstance(provider));
         }
 
+        private string GetDebuggerDisplay()
+        {
+            // For very big numbers, ToString can be too long or even timeout for Visual Studio to display
+            // Display an fast estimation value instead
+
+            // Use ToString for small values
+
+            if (_bits is null || _bits.Length <= 2)
+            {
+                return ToString();
+            }
+
+            // Estimate the value x as L*2^n, while L is the value of high bits, and n is the length of low bits
+            // Represent L as k*10^i, then x = L*2^n = k*10^(i+n*Log10(2))
+            // Let m=n*Log10(2), the final result would be x = (k*10^(m-[m]))*10^(i+[m])
+
+            ulong highBits = ((ulong)_bits[1] << kcbitUint) + _bits[0];
+            int lowBitsCount32 = _bits.Length - 2; // if Length > int.MaxValue/32, counting in bits can cause overflow
+            double exponentLow = lowBitsCount32 * kcbitUint * Math.Log10(2);
+
+            int exponent = (int)exponentLow;
+            double significand = (double)highBits * Math.Pow(10, exponentLow - exponent);
+
+            while (significand > 10.0)
+            {
+                significand /= 10.0;
+                exponent++;
+            }
+
+            string signStr = _sign == 0 ? "" : "-";
+
+            // Use about a half of the precision of double
+            return $"{signStr}{significand:F8}e+{exponent}";
```

Comment:
Maybe too much of an edge case, but `significand < 10.0` is not enough to guarantee `$"{significand:F8}"` is less than ten.

For example,
```C#
double d = 9.9999999999;
Console.WriteLine(d < 10.0);
Console.WriteLine(d.ToString("G17"));
Console.WriteLine(d.ToString("F8"));
```
outputs
```
True
9.9999999999
10.00000000
```

Reply:
We could instead multiply `significand` by some power of ten, round it to an integer, convert it into a decimal string, count the digits, add the decimal point, remove trailing zeros, ... , to avoid this problem.

Reply:
Rounding, dividing by power of ten, and then formatting in double would be sufficient.

Reply:
Combining with the FP precision problem, I think rounding instead of truncating looks better.